azurelinuxagent/common/utils/archive.py

```diff
@@ -76,19 +78,21 @@ def _get_files_to_archive(self):
 
         return files
 
-    def _archive(self, files):
+    def _archive(self, files, timestamp):
         for f in files:
-            shutil.move(f, self._target)
+            dst = os.path.join(self.history_dir(timestamp), os.path.basename(f))
+            shutil.move(f, dst)
 
     def _purge(self, files):
         for f in files:
             os.remove(f)
 
-    def _mkdir(self):
+    def _mkdir(self, timestamp):
         try:
-            fileutil.mkdir(self._target, mode=0o700)
+            d = self.history_dir(timestamp)
+            fileutil.mkdir(d, mode=0o700)
             return True
-        except:
+        except IOError as e:
             return False
```

Comment:
Is there a way to log e.strerror at this point? Otherwise we won't know why mkdir failed, and that might be useful.